cli/conf.py

```diff
@@ -53,7 +75,16 @@ def parse_args(cls, module_name, config, args=None):
         :param module_name: the module name: installer|provisioner|tester
         """
         cmd = clg.CommandLine(cls._get_specs(module_name, config))
-        return cmd.parse(args)
+        res_args = cmd.parse(args)
+
+        # override the command specific arguments from file
```

Comment:
This is exactly what def parse_args(...) is doing. 
If argument is defined in the cli then it will not be overridden by file. For example, 
```
# ir-provision virsh --host=myhost --from-file=provision.ini
# cat 
provision.ini
[virsh]
host=earth
```
This will resuts 'host' = myhost. 
I've updated utils.dict_merge function. It will merge keys only if value from the first dict is None.